src/Data/Nat/DivMod.agda

```diff
@@ -15,33 +15,34 @@ open import Data.Fin.Properties using (toℕ-fromℕ≤)
 open import Data.Nat as Nat
 open import Data.Nat.DivMod.Core
 open import Data.Nat.Divisibility.Core
+open import Data.Nat.Predicate
 open import Data.Nat.Properties
 open import Relation.Binary.PropositionalEquality
 open import Relation.Nullary.Decidable using (False)
 
 open ≤-Reasoning
 
 ------------------------------------------------------------------------
--- Definitions
+-- Definitions (with automatic non-zeroness)
 
 -- The division and modulus operations are only defined when the divisor
 -- is non-zero. The proof of this is defined as an irrelevant
--- implict argument of type `False (divisor ≟ 0)`. This allows this
--- proof to be automatically inferred when the divisor is of the form
--- `suc n`, and hence minimises the number of these proofs that
--- need be passed around. You can therefore write `m / suc n` without
--- issue.
+-- implict argument of type `NonZero divisor ≡ False (divisor ≟ 0)`.
```

Comment:
implict --> implicit

`NonZero divisor ≡ False (divisor ≟ 0)` looks like an equality type, probably not what you meant.  Expand to natural language, like 
> `NonZero divisor` which is defined as `False (divisor ≟ 0)` 
?!